data/helpers.d/php

```diff
@@ -2,32 +2,67 @@
 
 # Create a dedicated php-fpm config
 #
-# usage: ynh_add_fpm_config [--phpversion=7.X]
+# usage: ynh_add_fpm_config [--phpversion=7.X] [--dedicated_service]
 # | arg: -v, --phpversion - Version of php to use.
+# | arg: -d, --dedicated_service - Use a dedicated php-fpm service instead of the common one.
 #
 # Requires YunoHost version 2.7.2 or higher.
 ynh_add_fpm_config () {
 	# Declare an array to define the options of this helper.
-	local legacy_args=v
-	declare -Ar args_array=( [v]=phpversion= )
+	local legacy_args=vd
+	declare -Ar args_array=( [v]=phpversion= [d]=dedicated_service )
 	local phpversion
+	local dedicated_service
 	# Manage arguments with getopts
 	ynh_handle_getopts_args "$@"
 
 	# Configure PHP-FPM 7.0 by default
 	phpversion="${phpversion:-7.0}"
 
-	local fpm_config_dir="/etc/php/$phpversion/fpm"
-	local fpm_service="php${phpversion}-fpm"
+	# Do not use a dedicated service by default
+	dedicated_service=${dedicated_service:-0}
+
+	if [ $dedicated_service -eq 1 ]
+	then
+		local fpm_service="php${phpversion}-fpm-$app"
```

Comment:
I think we should really consider having the name of the dedicated fpm service just being `$app` ... Otherwise it's gonna be such a pain to remember how these are called when trying to help users. When we package nodejs apps (or whichever other techno than php) we usually just call the service `$app` because there's no point in calling it `nodejs10.3-$app`. 

It'd be so much simpler to just be able to run for example `systemctl status nextcloud` and `journalctl -u nextcloud` (and also seeing just 'nextcloud' in the webadmin service list)

```suggestion
		local fpm_service="$app"
```

Reply:
Mmh, at least something like $app-php
We're talking here about php, not the app itself.

Reply:
Depends what you mean by the app itself ... Again with nodejs apps and others, it's common practice to call the service $app and not $app-nodejs or $app-$tech :confused:

Reply:
Feel like node apps are more dependent to node, and thus to remove the node part in such apps is not a problem for me.
Php apps are dependent from nginx, php and fpm. So for me, considering that it's only the php-fpm part that we're talking about, and not all the app, for me would be better to have a service name that clearly said that it's only php.
Also, there's no such thing here as restarting the app. As you could do with etherpad for example.
If you change the config file for etherpad, you have to restart etherpad, and so node with it.
If you do so with wordpress, you don't have to restart anything, you will have to only if you change the php-fpm config for wordpress.

Reply:
```suggestion
		local fpm_service="${app}-phpfpm"
```
What about that ? Easy to find the service for an app, and quite clear about what it is.

Reply:
I vote for the last one!

Reply:
Hmf yea let's go for this  ¯\\\_\(ツ)\_/¯

Reply:
:stuck_out_tongue_closed_eyes: :trophy: